src/Compile.hs

```diff
@@ -39,15 +40,20 @@ data Result = Result
 printTime :: Args -> Compile.Result -> IO ()
 printTime Args {time} Compile.Result {compiledFile, duration} =
   when time $
-  putStrLn $ T.unpack $ T.pack compiledFile <> ": " <> T.pack (show duration)
+  putStrLn $
+  T.unpack $ T.pack compiledFile <> ": " <> T.pack (formatDuration duration)
+
+formatDuration :: Duration -> String
+formatDuration (Duration start end) = TL.unpack $ format timeSpecs start end
 
 data Duration = Duration
   { start :: TimeSpec
   , end :: TimeSpec
   }
 
 instance Show Duration where
-  show (Duration start end) = TL.unpack $ format timeSpecs start end
+  show (Duration start end) =
+    show (fromIntegral (toNanoSecs (diffTimeSpec end start)) / 1000000)
```

Comment:
Ah, so `show` is now only used to append to the log, and formats everything to msec. I get it! And the time printed to the terminal is still the old format, with different units.